src/java/org/jivesoftware/openfire/plugin/util/cache/ClusterListener.java

```diff
@@ -65,15 +63,24 @@ public class ClusterListener implements MembershipListener, LifecycleListener {
     private boolean clusterMember = false;
     private boolean isSenior;
 
-    ClusterListener(final Cluster cluster) {
-
+    synchronized void register(final Cluster cluster) {
+        logger.info("Cluster is being established. Executing queued events (if any).");
```

Comment:
I don't think this log message is clear. I think it's trying to say "Hazelcast cluster has been established - executing any queued events to complete Openfire cluster formation" ?

Reply:
Thanks, I write poor log statements, but having good ones could be very helpful.

I'd like to avoid wording that represents how the method is currently used (even if I can't see how it'd be used otherwise), so I'll replaced "Cluster is being established" with "Registering with cluster" and use the rest of your suggested sentence.